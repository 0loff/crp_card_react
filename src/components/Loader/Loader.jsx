import cls from './Loader.module.scss'

const Loader = props => {

    const {
        attrs = ''
    } = props;

    return (
        <div className={cls.loader_wrapper}>
            <div {...attrs} className={cls.loader}>
                <div></div>
                <div></div>
            </div>
        </div>
    )
}
export default Loader;
