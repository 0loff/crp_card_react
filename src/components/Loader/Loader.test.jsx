import React from 'react';
import Loader from './Loader';
import {render, screen} from "@testing-library/react";

describe('Loader component', () => {
    it('render Loader component', () => {
        render(<Loader attrs={{'data-testid': 'loader'}} />)

        expect(screen.getByTestId('loader')).toBeInTheDocument();
    })

    it('The Loader component is equal the snapshot', function () {
        const { asFragment } = render(<Loader />);

        expect(asFragment(<Loader />)).toMatchSnapshot();
    });
})