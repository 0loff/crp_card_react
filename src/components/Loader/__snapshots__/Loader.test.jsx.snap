// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component The Loader component is equal the snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader_wrapper"
  >
    <div
      class="loader"
    >
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
